docarray/array/storage/memory/seqlike.py

```diff
@@ -55,6 +57,6 @@ def __add__(self, other: Union['Document', Sequence['Document']]):
         v.extend(other)
         return v
 
+    @needs_id2offset_rebuild
     def extend(self, values: Iterable['Document']) -> None:
         self._data.extend(values)
-        self._rebuild_id2offset()
```

Comment:
I think extending doesn't require rebuilding because we just need to add more offsets

Reply:
Done in the last commit